### PR TITLE
bpo-40094: Enhance os.WIFEXITED documentation

### PR DESCRIPTION
**Pull request by vstinner**
https://bugs.python.org/issue40094

### CODE REVIEW

**Comment by vstinner** [comment]
@JulienPalard @deronnax: Would you mind to review this change? I used "man 2 waitpid" on my Fedora and combined it with the existing documentation.

Is it ok to backport such documentation enhancement to 3.7 and 3.8?

I was unhappy with WIFEXITED and WIFSIGNALED documentation. The difference between the two was unclear. I wrote this PR while working on PR #19201 documentation.

File: Doc/library/os.rst [hunk 1]
```diff
@@ -4167,45 +4167,55 @@ used to determine the disposition of a process.
    Return ``True`` if a core dump was generated for the process, otherwise
    return ``False``.
 
+   This function should be employed only if :func:`WIFSIGNALED` returned true.
```
Comment (corona10) [inline Doc/library/os.rst]:
```suggestion
   This function should be employed only if :func:`WIFSIGNALED` returned ``True``.
```

Comment (corona10) [reply Doc/library/os.rst]: Same for others what do you think?

Comment (vstinner) [reply Doc/library/os.rst]: Hum, I copied the code from the manual page which is written for C code. I rephased it to:

```
only if :func:`WIFSIGNALED` is true
```

**Comment by vstinner** [comment]
cc @pablogsal

**Review by corona10 — APPROVED** [review]
lgtm :)

**Pull request merged**